pymc3/distributions/multivariate.py

```diff
@@ -687,6 +688,32 @@ def _distr_parameters_for_repr(self):
         return ["n", "a"]
 
 
+class OrderedMultinomial(Multinomial):
+    rv_op = multinomial
+
+    @classmethod
+    def dist(cls, eta, cutpoints, n, compute_p=True, *args, **kwargs):
+        eta = at.as_tensor_variable(floatX(eta))
+        cutpoints = at.as_tensor_variable(cutpoints)
+        n = at.as_tensor_variable(n)
+
+        pa = sigmoid(cutpoints - at.shape_padright(eta))
+        p_cum = at.concatenate(
+            [
+                at.zeros_like(at.shape_padright(pa[..., 0])),
+                pa,
+                at.ones_like(at.shape_padright(pa[..., 0])),
+            ],
+            axis=-1,
+        )
+        if compute_p:
+            p = pm.Deterministic("complete_p", p_cum[..., 1:] - p_cum[..., :-1])
```

Comment:
This should probably check if it's being called inside a model context, otherwise it will raise a perhaps not very informative error. I.e., when calling `OrderedMultinomial.dist(...)` outside of a `Model` context.

Reply:
Also name should extend the user defined name. Otherwise you'll get an error when using two distributions inside a model due to the repeatedly Deterministic name.

Reply:
Good point! How do I get the user defined name in v4?

Reply:
It's one of the optional kwargs

Reply:
I've tried a few things but can't figure it out 🤔

Reply:
Then it's in `__new__` I guess

Reply:
You mean `cls.__new__().name` ?